services/database/programcardinfo.ts

```diff
@@ -0,0 +1,168 @@
+import prisma from "@database";
+import { ProgramCardInfo } from "models/Program";
+import { ClassCardInfo } from "models/Class";
+import { locale } from "@prisma/client";
+
+/**
+ * @param {string} id - programId
+ */
+async function getClassInfoWithProgramId(id: string): Promise<ClassCardInfo[]> {
+    const findResults = await prisma.class.findMany({
+        where: {
+            programId: parseInt(id),
+            isArchived: false,
+        },
+        select: {
+            ageGroup: true,
+            spaceTotal: true,
+            id: true,
+            name: true,
+            spaceAvailable: true,
+            volunteerSpaceTotal: true,
+            volunteerSpaceAvailable: true,
+            startDate: true,
+            endDate: true,
+            weekday: true,
+            startTimeMinutes: true,
+            durationMinutes: true,
+            imageLink: true,
+
+            classTranslation: {
+                where: {
+                    language: locale.en, // TODO dont hardcode language
+                },
+                select: {
+                    name: true,
+                    description: true,
+                },
+            },
+            teacherRegs: {
+                select: {
+                    teacher: {
+                        select: {
+                            user: true,
+                        },
+                    },
+                },
+            },
+        },
+    });
+    return findResults.map((result) => {
```

Comment:
What I remember doing last term is that we had util files which would do this mapping for converting the data into what we wanted. In database, we returned everything but we when used the function in the util file to clean it up a bit.

For example, in this case, we would first getClassByProgramId. Then, we would call this util function in the file in `api` to get the data in the format we want it to be. 

We could also do the same here, it would just be moving where you're mapping to a file within `util`. I think this would make the functions here a lot more usable if it returns all the information